From 5425c16cc9394eea141d07c13e43b4ae9266cd9d Mon Sep 17 00:00:00 2001
From: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
Date: Mon, 25 Jun 2018 00:14:40 -0700
Subject: [PATCH 1229/1240] ASoC: SOF: merge sample_bits tokens with the rest
 of the ssp_tokens

Sample bits is SSP specific. So move it to be part of SSP
tokens. dai_ssp_link_tokens are not need anymore. So remove it
and the corresponding call to parse the tokens.

Signed-off-by: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
---
 include/uapi/sound/sof-topology.h |  2 +-
 sound/soc/sof/topology.c          | 18 +++---------------
 2 files changed, 4 insertions(+), 16 deletions(-)

diff --git a/include/uapi/sound/sof-topology.h b/include/uapi/sound/sof-topology.h
index 721a2b5..096aaed 100644
--- a/include/uapi/sound/sof-topology.h
+++ b/include/uapi/sound/sof-topology.h
@@ -37,7 +37,6 @@
 #define	SOF_TKN_DAI_DMAC_CONFIG			153
 #define SOF_TKN_DAI_TYPE			154
 #define SOF_TKN_DAI_INDEX			155
-#define SOF_TKN_DAI_SAMPLE_BITS			156
 
 /* scheduling */
 #define SOF_TKN_SCHED_DEADLINE			200
@@ -69,6 +68,7 @@
 #define SOF_TKN_INTEL_SSP_BCLK_KEEP_ACTIVE	501
 #define SOF_TKN_INTEL_SSP_FS_KEEP_ACTIVE	502
 #define SOF_TKN_INTEL_SSP_MCLK_ID		503
+#define SOF_TKN_INTEL_SSP_SAMPLE_BITS		504
 
 /* DMIC */
 #define SOF_TKN_INTEL_DMIC_DRIVER_VERSION	600
diff --git a/sound/soc/sof/topology.c b/sound/soc/sof/topology.c
index 1d4685c..f934738 100644
--- a/sound/soc/sof/topology.c
+++ b/sound/soc/sof/topology.c
@@ -336,11 +336,6 @@ static const struct sof_topology_token dai_link_tokens[] = {
 		offsetof(struct sof_ipc_dai_config, type), 0},
 };
 
-static const struct sof_topology_token dai_ssp_link_tokens[] = {
-	{SOF_TKN_DAI_SAMPLE_BITS, SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32,
-		offsetof(struct sof_ipc_dai_ssp_params, sample_valid_bits), 0},
-};
-
 /* scheduling */
 static const struct sof_topology_token sched_tokens[] = {
 	{SOF_TKN_SCHED_DEADLINE, SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32,
@@ -414,6 +409,9 @@ static const struct sof_topology_token ssp_tokens[] = {
 	{SOF_TKN_INTEL_SSP_MCLK_ID,
 		SND_SOC_TPLG_TUPLE_TYPE_SHORT, get_token_u16,
 		offsetof(struct sof_ipc_dai_ssp_params, mclk_id), 0},
+	{SOF_TKN_INTEL_SSP_SAMPLE_BITS, SND_SOC_TPLG_TUPLE_TYPE_WORD,
+		get_token_u32,
+		offsetof(struct sof_ipc_dai_ssp_params, sample_valid_bits), 0},
 };
 
 /* DMIC */
@@ -1473,16 +1471,6 @@ static int sof_link_ssp_load(struct snd_soc_component *scomp, int index,
 	memset(&config->ssp, 0, sizeof(struct sof_ipc_dai_ssp_params));
 	config->hdr.size = size;
 
-	/* get any bespoke DAI tokens */
-	ret = sof_parse_tokens(scomp, &config->ssp, dai_ssp_link_tokens,
-			       ARRAY_SIZE(dai_ssp_link_tokens),
-			       private->array, private->size);
-	if (ret != 0) {
-		dev_err(sdev->dev, "error: parse ssp link tokens failed %d\n",
-			private->size);
-		return ret;
-	}
-
 	ret = sof_parse_tokens(scomp, &config->ssp, ssp_tokens,
 			       ARRAY_SIZE(ssp_tokens), private->array,
 			       private->size);
-- 
2.7.4

